app/src/lib/components/CommitList.svelte

```diff
@@ -97,8 +107,31 @@
 		}
 		branchController.insertBlankCommit($branch.id, commitId, location === 'above' ? -1 : 1);
 	}
+
+	function getReorderDropzoneOffset({
+		isFirst = false,
+		isMiddle = false,
+		isLast = false
+	}: {
+		isFirst?: boolean;
+		isMiddle?: boolean;
+		isLast?: boolean;
+	}) {
+		if (isFirst) return 12;
+		if (isMiddle) return 6;
+		if (isLast) return 0;
+		return 0;
+	}
 </script>
 
+{#snippet reorderDropzone(dropzone: ReorderDropzone, yOffsetPx: number)}
+	<Dropzone accepts={dropzone.accepts.bind(dropzone)} ondrop={dropzone.onDrop.bind(dropzone)}>
+		{#snippet overlay({ hovered, activated })}
+			<LineOverlay {hovered} {activated} {yOffsetPx} />
+		{/snippet}
+	</Dropzone>
+{/snippet}
```

Comment:
I didn't think an entire component was required for the Reordering dropzone. I'm willing to be swayed on this position though.